index.js

```diff
@@ -85,81 +91,113 @@ class GithubScm extends Scm {
-                }
+                params: { id: scmUriParts[scmId] }
             }, (error, data) => {
                 if (error) {
                     return reject(error);
                 }
 
-                return resolve(data.permissions);
+                const repoNameParts = data.full_name.split('/');
+
+                return resolve({
+                    branch: scmUriParts[PART_SCM_URI_BRANCH],
+                    host: scmUriParts[PART_SCM_URI_HOST],
+                    repo: repoNameParts[PART_REPO_NAME],
+                    user: repoNameParts[PART_OWNER_NAME]
+                });
             });
         });
     }
 
     /**
+    * Get a users permissions on a repository
+    * @method _getPermissions
+    * @param  {Object}   config            Configuration
+    * @param  {String}   config.scmUri     The scmUri to get permissions on
+    * @param  {String}   config.token      The token used to authenticate to the SCM
+    * @return {Promise}
+    */
+    _getPermissions(config) {
+        return this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        }).then(scmInfo =>
+            new Promise((resolve, reject) => {
+                this.breaker.runCommand({
+                    action: 'get',
+                    token: config.token,
+                    params: {
+                        repo: scmInfo.repo,
+                        user: scmInfo.user
+                    }
+                }, (error, data) => {
+                    if (error) {
+                        return reject(error);
+                    }
+
+                    return resolve(data.permissions);
+                });
+            })
+        );
+    }
+
+    /**
      * Get a commit sha for a specific repo#branch
      * @method getCommitSha
      * @param  {Object}   config            Configuration
-     * @param  {String}   config.scmUrl     The scmUrl to get commit sha of
+     * @param  {String}   config.scmUri     The scmUri to get commit sha of
      * @param  {String}   config.token      The token used to authenticate to the SCM
      * @return {Promise}
      */
     _getCommitSha(config) {
-        const scmInfo = getInfo(config.scmUrl);
-
-        return new Promise((resolve, reject) => {
-            this.breaker.runCommand({
-                action: 'getBranch',
-                token: config.token,
-                params: scmInfo
-            }, (error, data) => {
-                if (error) {
-                    return reject(error);
-                }
-
-                return resolve(data.commit.sha);
-            });
-        });
+        return this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        }).then(scmInfo =>
+            new Promise((resolve, reject) => {
+                this.breaker.runCommand({
+                    action: 'getBranch',
+                    token: config.token,
+                    params: {
```

Comment:
Nah, that's too much. If `lookupScmUri` changes its output, then there's no need to have this function be affected also. This is a defensive and deterministic way as to what we're passing to `.runCommand`.